packages/components/src/components/dropdown/dropdown.js

```diff
@@ -91,53 +93,125 @@ class Dropdown extends mixin(
-          this.element.focus();
         }
       });
+
       const listItems = toArray(
         this.element.querySelectorAll(this.options.selectorItem)
       );
-      listItems.forEach(item => {
-        if (this.element.classList.contains(this.options.classOpen)) {
-          item.tabIndex = 0;
-        } else {
-          item.tabIndex = -1;
+      // only want to grab the listNode IF it's using the latest a11y HTML structure
+      const listNode = triggerNode
+        ? this.element.querySelector(this.options.selectorMenu)
+        : null;
+
+      // @todo remove conditionals for elements existing once legacy structure is depreciated
+      if (
+        changedState &&
+        this.element.classList.contains(this.options.classOpen)
+      ) {
+        // toggled open
+        if (triggerNode) {
+          triggerNode.setAttribute('aria-expanded', 'true');
         }
-      });
+        (listNode || this.element).focus();
+        if (listNode) {
+          const selectedNode = listNode.querySelector(
+            this.options.selectorItemSelected
+          );
+          listNode.setAttribute(
+            'aria-activedescendant',
+            (selectedNode || listItems[0]).id
+          );
+          (selectedNode || listItems[0]).classList.add(
+            this.options.classFocused
+          );
+        }
+      } else if (changedState && (isOfSelf || actions.remove)) {
+        // toggled close
+        (triggerNode || this.element).focus();
+        if (triggerNode) {
+          triggerNode.setAttribute('aria-expanded', 'false');
+        }
+        if (listNode) {
+          listNode.removeAttribute('aria-activedescendant');
+          this.element
+            .querySelector(this.options.selectorItemFocused)
+            .classList.remove(this.options.classFocused);
+        }
+      }
+
+      // @todo remove once legacy structure is depreciated
+      if (!triggerNode) {
+        listItems.forEach(item => {
+          if (this.element.classList.contains(this.options.classOpen)) {
+            item.tabIndex = 0;
+          } else {
+            item.tabIndex = -1;
+          }
+        });
+      }
     }
   }
 
   /**
    * @returns {Element} Currently highlighted element.
    */
   getCurrentNavigation() {
-    const focused = this.element.ownerDocument.activeElement;
-    return focused.nodeType === Node.ELEMENT_NODE &&
-      focused.matches(this.options.selectorItem)
-      ? focused
-      : null;
+    let focusedNode;
+
+    // Using the latest semantic markup structure where trigger is a button
+    // @todo remove conditional once legacy structure is depreciated
+    if (this.element.querySelector(this.options.selectorTrigger)) {
+      const listNode = this.element.querySelector(this.options.selectorMenu);
+      const focusedId = listNode.getAttribute('aria-activedescendant');
+      focusedNode = focusedId ? listNode.querySelector(`#${focusedId}`) : null;
```

Comment:
`getElementById()`?

Reply:
@asudoh `getElementById` only works from the `document` object which is why I opted for `querySelector` here.  I do believe that I could use `this.element.ownerDocument.getElementById` here though if you'd prefer that.

Reply:
OK thank you for clarifying @elizabethsjudd! - Makes sense as-is.